### PR TITLE
update version for 4.1.0 release

### PR DESCRIPTION
**Pull request by yiliuTo**
This pr is to udpate the sample dependency and README version for 4.1.0 release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,66 +19,67 @@ We use `main` branch as the develop branch while setting the default branch as t
 |---------------------------|----------------|----------------|
 | main                      | false          | develop branch |
 | spring-cloud-azure_v4.0.0 | true           | release branch |
```
Comment (chenrujun) [inline README.md]: This is not default branch.

Comment (saragluna) [inline README.md:21]: we should delete this one of v4.0.0

File: README.md [hunk 2]
```diff
@@ -19,66 +19,67 @@ We use `main` branch as the develop branch while setting the default branch as t
...
+- [spring-cloud-dependencies:2021.0.2](https://repo.maven.apache.org/maven2/org/springframework/cloud/spring-cloud-dependencies/2021.0.2/spring-cloud-dependencies-2021.0.2.pom).
 
 ## All Active Branches
 
-| Spring Boot Version  | Spring Cloud version  | Spring Cloud Azure Version                                                                 |
-|----------------------|-----------------------|--------------------------------------------------------------------------------------------|
-| 2.6.3                | 2021.0.1              | [4.0.0](https://github.com/Azure/azure-sdk-for-java/releases/tag/spring-cloud-azure_4.0.0) | 
+| Spring Boot Version | Spring Cloud version | Spring Cloud Azure Version                                                                 |
+|---------------------|----------------------|--------------------------------------------------------------------------------------------|
+| 2.6.3               | 2021.0.1             | [4.0.0](https://github.com/Azure/azure-sdk-for-java/releases/tag/spring-cloud-azure_4.0.0) |
```
Comment (saragluna) [inline README.md:32]: same here

File: README.md [hunk 3]
```diff
@@ -19,66 +19,67 @@ We use `main` branch as the develop branch while setting the default branch as t
...
-
+| spring-cloud-azure_v4.1.0 | true           | release branch |
 
 ## Current Branch Supported Versions
-- [spring-boot-dependencies:2.6.3](https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/2.6.3/spring-boot-dependencies-2.6.3.pom).
-- [spring-cloud-dependencies:2021.0.1](https://repo.maven.apache.org/maven2/org/springframework/cloud/spring-cloud-dependencies/2021.0.1/spring-cloud-dependencies-2021.0.1.pom).
+- [spring-boot-dependencies:2.6.7](https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/2.6.7/spring-boot-dependencies-2.6.7.pom).
+- [spring-cloud-dependencies:2021.0.2](https://repo.maven.apache.org/maven2/org/springframework/cloud/spring-cloud-dependencies/2021.0.2/spring-cloud-dependencies-2021.0.2.pom).
 
 ## All Active Branches
```
Comment (chenrujun) [inline README.md:27]: Active branch means branch in `azure-spring-boot-samples`

**Review by saragluna — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -19,66 +19,67 @@ We use `main` branch as the develop branch while setting the default branch as t
...
 
-| Spring Boot Version  | Spring Cloud version  | Spring Cloud Azure Version                                                                 |
-|----------------------|-----------------------|--------------------------------------------------------------------------------------------|
-| 2.6.3                | 2021.0.1              | [4.0.0](https://github.com/Azure/azure-sdk-for-java/releases/tag/spring-cloud-azure_4.0.0) | 
+| Spring Boot Version | Spring Cloud version | Spring Cloud Azure Version                                                                 |
+|---------------------|----------------------|--------------------------------------------------------------------------------------------|
+| 2.6.3               | 2021.0.1             | [4.0.0](https://github.com/Azure/azure-sdk-for-java/releases/tag/spring-cloud-azure_4.0.0) | 
+| 2.6.7               | 2021.0.2             | [4.1.0](https://github.com/Azure/azure-sdk-for-java/releases/tag/spring-cloud-azure_4.1.0) |
```
Comment (chenrujun) [inline README.md:31]: It's necessary to add a new column about `azure-spring-boot-samples` branches:
image

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (yiliuTo) [reply README.md]: Got it

File: README.md (hunk 4 above)
Comment (chenrujun) [reply README.md:31]: We can delete this table if we only have 1 active branch.

**Comment by yiliuTo** [comment]
hey @saragluna  and @chenrujun I would like to verify the branch policy of the `main` branch and discuss whether we should remain non-latest version in README of main branch.

 As is documented, main branch is used for developing, and I suspect it always targets to updating our samples towards the latest version? Thus I can delete the section of [all active branches](https://github.com/Azure-Samples/azure-spring-boot-samples/tree/main#all-active-branches) given the information has mentioned [above](https://github.com/Azure-Samples/azure-spring-boot-samples/tree/main#branch-policy). Also I should only present main and the latest release branch info in the secion [branch Policy](https://github.com/Azure-Samples/azure-spring-boot-samples/tree/main#branch-policy)? Do you have any insights?

**Comment by yiliuTo** [comment]
Since we will only reserve one version branch for our latest release, thus we will remove information about 4.0.0 in main branch.

**Review by chenrujun — APPROVED** [review]
LGTM.

**Review by saragluna — APPROVED** [review]

**Pull request merged**